cdap-security/src/main/java/io/cdap/cdap/security/auth/context/AuthenticationContextModules.java

```diff
@@ -149,10 +162,35 @@ public Module getNoOpModule() {
       @Override
       protected void configure() {
         bind(AuthenticationContext.class).to(AuthenticationTestContext.class);
+        bind(AuthenticationContext.class).annotatedWith(Names.named(RemoteClientFactory.REMOTE_CLIENT))
+          .toProvider(RemoteClientAuthenticationContextProvider.class);
       }
     };
   }
 
+  /**
+   * A {@link Provider} for {@link AuthenticationContext} for use in the
+   * {@link io.cdap.cdap.common.internal.remote.RemoteClient}.
+   */
+  private static final class RemoteClientAuthenticationContextProvider implements Provider<AuthenticationContext> {
+    private final CConfiguration cConf;
+    private final Injector injector;
+
+    @Inject
+    RemoteClientAuthenticationContextProvider(CConfiguration cConf, Injector injector) {
+      this.cConf = cConf;
+      this.injector = injector;
+    }
+
+    @Override
+    public AuthenticationContext get() {
+      if (SecurityUtil.isInternalAuthEnabled(cConf)) {
+        return injector.getInstance(AuthenticationContext.class);
+      }
+      return null;
```

Comment:
instead of using `null`, use an "empty" implementation of `AuthenticationContext`

Reply:
Fixed